runtime/README.md

```diff
@@ -692,107 +708,504 @@ Command Code: `0x434B_4558` ("CKEX")
+| plaintext size | u32      | Can be 0 if this is not the final message |
+| plaintext      | u8[...]  |                                           |
+
+The encrypted and authenticated context's internal structure will be:
+
+*Table: `CM_AES_GCM_DECRYPT` internal context*
+| **Name**        | **Type** | **Description** |
+| --------------- | -------- | --------------- |
+| tag size        | u32      |                 |
+| last length     | u32      |                 |
+| last iv         | u[16]    |                 |
+| last GHASH      | u[16]    |                 |
+| last ciphertext | u[16]    |                 |
+
+
+### CM_ECDH_GENERATE
+
+This computes the first half of an Elliptic Curve Diffie-Hellman exchange.
+
+Currently only supports the NIST P-384 curve.
+
+The returned context must be passed to the `CM_ECDH_FINISH` command.
+
+Command Code: `0x434D_4547` ("CMEG")
+
+*Table: `CM_ECDH_GENERATE` input arguments*
+| **Name**    | **Type** | **Description**      |
+| ----------- | -------- | -------------------- |
+| chksum      | u32      |                      |
+| curve/flags | u32      | Must be 0. Reserved. |
+
+*Table: `CM_ECDH_GENERATE` output arguments*
+| **Name**           | **Type** | **Description**                      |
+| ------------------ | -------- | ------------------------------------ |
+| chksum             | u32      |                                      |
+| fips_status        | u32      | FIPS approved or an error            |
+| context size       | u32      | size of context                      |
+| context            | u8[...]  |                                      |
+| exchange data size | u32      | size of data to send in the exchange |
+| exchange data      | u8[...]  | i.e., the public point               |
+
+### CM_ECDH_FINISH
+
+This computes the second half of an Elliptic Curve Diffie-Hellman exchange.
+
+Currently only supports the NIST P-384 curve.
+
+The context must be passed from the `CM_ECDH_GENERATE` command.
+
+Command Code: `0x434D_4546` ("CMEF")
+
+*Table: `CM_ECDH_FINISH` input arguments*
+| **Name**                    | **Type** | **Description**               |
+| --------------------------- | -------- | ----------------------------- |
+| chksum                      | u32      |                               |
+| context size                | u32      | size of context               |
+| context                     | u8[...]  |                               |
+| incoming exchange data size | u32      |                               |
+| incoming exchange data      | u8[...]  | the other side's public point |
+
+*Table: `CM_ECDH_FINISH` output arguments*
+| **Name**    | **Type** | **Description**                  |
+| ----------- | -------- | -------------------------------- |
+| chksum      | u32      |                                  |
+| fips_status | u32      | FIPS approved or an error        |
+| output CMID | u8[32]   | Output CMID of the shared secret |
```

Comment:
What kind of key is this, behind the CMID? Seems like it should be something that I can only pass to `CM_HKDF`.

Reply:
This will create a 384-bit shared secret. Theoretically it should only be used by HKDF, but other protocols could use the shared secret directly for AES.

Reply:
We should document the secret size. And document that the AES commands will implicitly truncate the CMID key if it is too large.